src/components/MenuPanel/MenuPanel.js

```diff
@@ -69,6 +69,7 @@ function MenuPanel({
 
   const handleToggleSystemApps = useCallback(() => {
     setSystemAppsOpened(opened => !opened)
+    systemAppsOpenedState.set(!systemAppsOpenedState.isOpen())
```

Comment:
Ahh good catch! But we probably want to be doing this in a separate `useEffect` based on the `systemAppsOpened` state.

Reply:
Sure!

Reply:
Hmm, actually, the reason it's written like this is because we only want to set the opened state when the user toggles it manually. If we want to make this reactive to the `systemAppsOpened` state, then for automatically opening the system apps menu, we would have to introduce another variable like `isManuallyToggled`, so we can remember if the user actually toggled it.